### ISSUE TITLE
Run once and reuse?

### ISSUE THREAD

**Issue by Explosion-Scratch** [Skyvern-AI/skyvern#1062]
The most useful application of Skyvern imo would be to have it run once (or once in a while) then reuse the workflow it generates. Could it be made to generate puppeteer code? Or some sort of YAML file that doesn't need AI to run again?

This would help significantly to cut costs and have more reliability while still benefiting from the AI integration

**Comment by cdelgadob** [comment]
That my opinion as well. I was thinking about the AI functionality as kind of a fallback mechanism. So it would be: run once and validate. Then run and if it fails use the AI again to adapt to possible changes.

Or you could check if the page has changed before running the agent using some sort of cache.

What do you think?



Enviado desde Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: --Explosion-- ***@***.***>
Enviado: Saturday, October 26, 2024 7:52:21 PM
Para: Skyvern-AI/skyvern ***@***.***>
Cc: Subscribed ***@***.***>
Asunto: [Skyvern-AI/skyvern] Run once and reuse? (Issue #1062)


The most useful application of Skyvern imo would be to have it run once (or once in a while) then reuse the workflow it generates. Could it be made to generate puppeteer code? Or some sort of YAML file that doesn't need AI to run again?

This would help significantly to cut costs and have more reliability while still benefiting from the AI integration

**Comment by suchintan** [comment]
We have this in beta right now. If you create a task block inside workflows, there's a "cache" option under advanced

Cached actions don't use LLMs under the hood!

**Comment by cdelgadob** [comment]
That’s good news! And what happens if the task fails? Does it fallback to LLM to retry?

Enviado desde Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: Suchintan ***@***.***>
Enviado: Saturday, October 26, 2024 10:54:54 PM
Para: Skyvern-AI/skyvern ***@***.***>
Cc: Carlos Delgado ***@***.***>; Comment ***@***.***>
Asunto: Re: [Skyvern-AI/skyvern] Run once and reuse? (Issue #1062)


We have this in beta right now. If you create a task block inside workflows, there's a "cache" option under advanced

Cached actions don't use LLMs under the hood!

**Comment by suchintan** [comment]
Yeah exactly. If the cache fails, it falls back to the LLM and adds the successful thing to the cache

**Comment by Explosion-Scratch** [comment]
> We have this in beta right now. If you create a task block inside workflows, there's a "cache" option under advanced
> 
> Cached actions don't use LLMs under the hood!

Does the cache option allow different inputs each time? Or does it just run the exact same thing again? E.g. filling out the same form with different values. Also is there a way to export and import this?

**Comment by suchintan** [comment]
1. It allows for different inputs
2. If you go to the workflows page, you should see the option to export a specific workflow as YAML or JSON — is that what you're looking for?
